sqlglot/dialects/hive.py

```diff
@@ -18,6 +18,33 @@
 from sqlglot.parser import Parser, parse_var_map
 from sqlglot.tokens import Tokenizer
 
+# (FuncType, Multiplier)
+DATE_DELTA_INTERVAL = {
+    "YEAR": ("ADD_MONTHS", 12),
+    "MONTH": ("ADD_MONTHS", 1),
+    "QUARTER": ("ADD_MONTHS", 3),
+    "WEEK": ("DATE_ADD", 7),
+    "DAY": ("DATE_ADD", 1),
+}
+
+
+def _add_date_sql(self, expression):
+    unit = expression.text("unit").upper()
+    func, multiplier = DATE_DELTA_INTERVAL.get(unit, ("DATE_ADD", 1))
+    modified_increment = (
+        int(expression.text("expression")) * multiplier if expression.expression.is_number else expression.expression
+    )
+    return f"{func}({self.sql(expression, 'this')}, {modified_increment})"
```

Comment:
all funcs should use self.format_args(expression.this, modified_increment)

Reply:
`   python 
return f"{func}({self.format_args(e.text('unit'), e.expression, e.this)})"
`
Is that ok with the e.text?

Reply:
it might be but you can just do e.args.get("unit")

Reply:
It takes the quotes that way I thought, which would be incorrect for TSQL.

Reply:
ran quick test:
![image](https://user-images.githubusercontent.com/18576405/197595273-807f3066-5ce5-418f-b9ed-1927507291e1.png)